src/fvdb/detail/utils/SimpleOpHelper.h

```diff
@@ -0,0 +1,133 @@
+// Copyright Contributors to the OpenVDB Project
+// SPDX-License-Identifier: Apache-2.0
+//
+#ifndef FVDB_DETAIL_UTILS_SIMPLEOPHELPER_H
+#define FVDB_DETAIL_UTILS_SIMPLEOPHELPER_H
+
+#include <fvdb/detail/GridBatchImpl.h>
+#include <fvdb/detail/utils/AccessorHelpers.cuh>
+#include <fvdb/detail/utils/ForEachCPU.h>
+#include <fvdb/detail/utils/cuda/ForEachCUDA.cuh>
+#include <fvdb/detail/utils/cuda/ForEachPrivateUse1.cuh>
+
+#include <c10/cuda/CUDAException.h>
+
+#include <cuda_runtime.h>
+
+#include <array>
+
+namespace fvdb {
+namespace detail {
+
+template <torch::DeviceType DeviceTag, typename T, size_t... TailDims>
+struct OutAccessorTypeHelper {
+    static constexpr size_t N = sizeof...(TailDims) + 1;
+    using type = decltype(tensorAccessor<DeviceTag, T, N>(std::declval<torch::Tensor &>()));
+    type accessor;
+};
+
+template <torch::DeviceType DeviceTag, typename T, size_t... TailDims>
+using OutAccessor_t = typename OutAccessorTypeHelper<DeviceTag, T, TailDims...>::type;
+
+template <torch::DeviceType DeviceTag> struct DeviceValidationHelper;
+
+template <> struct DeviceValidationHelper<torch::kCPU> {
+    static void
+    validate(torch::Device const &device) {
+        TORCH_CHECK(device.is_cpu(), "Device must be CPU when DeviceTag is torch::kCPU");
+    }
+};
+
+template <> struct DeviceValidationHelper<torch::kCUDA> {
+    static void
+    validate(torch::Device const &device) {
+        TORCH_CHECK(device.is_cuda(), "Device must be CUDA when DeviceTag is torch::kCUDA");
+    }
+};
+
+template <> struct DeviceValidationHelper<torch::kPrivateUse1> {
+    static void
+    validate(torch::Device const &device) {
+        TORCH_CHECK(device.is_privateuseone(),
+                    "Device must be PrivateUse1 when DeviceTag is torch::kPrivateUse1");
+    }
+};
+
+template <torch::DeviceType DeviceTag>
+void
+validate_device(torch::Device const &device) {
```

Comment:
I know we're not 100% consistent everywhere but we have been trying to use camelCase for everything including function names